lib/ledger_sync/adaptors/quickbooks_online/webhook_event.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# ref: https://developer.intuit.com/app/developer/qbo/docs/develop/webhooks/managing-webhooks-notifications#validating-the-notification
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      class WebhookEvent
+        attr_reader :deleted_id,
+                    :event_operation,
+                    :last_updated_at,
+                    :ledger_id,
+                    :original_payload,
+                    :payload,
+                    :quickbooks_online_resource_type,
+                    :webhook_notification,
+                    :webhook
+
+        def initialize(payload:, webhook_notification: nil)
+          @original_payload = payload
+          payload = JSON.parse(payload) if payload.is_a?(String)
+          @payload = payload
+
+          @deleted_id = payload.dig('deletedId')
+
+          @event_operation = payload.dig('operation')
+          raise 'Invalid payload: Could not find operation' if @event_operation.blank?
+
+          @last_updated_at = payload.dig('lastUpdated')
+          raise 'Invalid payload: Could not find lastUpdated' if @last_updated_at.blank?
+
+          @last_updated_at = Time.parse(@last_updated_at)
+
+          @ledger_id = payload.dig('id')
+          raise 'Invalid payload: Could not find id' if @ledger_id.blank?
+
+          @quickbooks_online_resource_type = payload.dig('name')
+          raise 'Invalid payload: Could not find name' if @quickbooks_online_resource_type.blank?
+
+          @webhook_notification = webhook_notification
+          @webhook = webhook_notification.try(:webhook)
+        end
+
+        def find(adaptor:)
+          adaptor.class.base_operation_module_for(resource_class: resource_class)::Find.new(
```

Comment:
Can you do a check if the Find operation exists, as well as if the base operations exist? I imagine we'll get webhooks for some things we don't have resources for

Reply:
True.  How do you want to handle that downstream if it doesn't exist?

Reply:
Hmm. Good question. We could have a class like `UnsupportedResource` and just return that with the ledger_id on it. That way you can iterate past it.
The other way is to return `nil` here and then `compact` (optionally) where we get the events.

Reply:
I actually made some updates.  Let me know what you think.  I think this is clearer, letting `.resource` to return a local resource with just the `ledger_id` set.  Then there are helper methods for performing the finds yourself if you want.

We still have the issue that it will error if `resource_class` does not exist (e.g. an object we haven't created yet).  For now, I have `resource` returning `nil` if that's the case.